node/buildpack.go

```diff
@@ -35,23 +35,21 @@ func (bp *NodeBuildpack) Id() string {
 	return bp.id
 }
 
-func (bp *NodeBuildpack) Detect(d detect.Detect, m function.Metadata) (*buildplan.BuildPlan, error) {
+func (bp *NodeBuildpack) Detect(d detect.Detect, m function.Metadata) (*buildplan.Plan, error) {
 	if detected, err := bp.detect(d, m); err != nil {
 		return nil, err
 	} else if detected {
-		plan := BuildPlanContribution(d, m)
+		plan := BuildPlanContribution(m)
 		return &plan, nil
 	}
 	// didn't detect
 	return nil, nil
 }
 
 func (*NodeBuildpack) detect(d detect.Detect, m function.Metadata) (bool, error) {
-	// Try npm
-	if _, ok := d.BuildPlan[modules.Dependency]; ok {
+	if ok, _ := DetectNpm(d, modules.Dependency); ok {
```

Comment:
This change is wrong (and the related test change as well, obviously).
I think the correct way is to declare "node_modules" as a required dependency in `BuildPlanContribution` (see below), `npm-cnb` contributes it.